junebug/tests/test_logging_service.py

```diff
@@ -0,0 +1,197 @@
+            # json encoding changes all tuples to lists
+            'stack': json.loads(json.dumps(f.stack)),
+        })
+
+    def test_log_warning(self):
+        '''Logging an warning level log should generate the correct level log
+        message'''
+        self.obs({'message': ["a"], 'system': 'foo',
+                  'logLevel': logging.WARN})
+        [log] = self.logfile.logs
+        log = json.loads(log)
+        log.pop('timestamp')
+
+        self.assertEqual(log, {
+            'level': logging.WARN,
+            'logger': 'worker-1.foo',
+            'message': 'a',
+        })
+
+    def test_log_info(self):
+        '''Logging an info level log should generate the correct level log
+        message'''
+        self.obs({'message': ["a"], 'system': 'test.log'})
+        [log] = self.logfile.logs
+        log = json.loads(log)
+        log.pop('timestamp')
+
+        self.assertEqual(log, {
+            'logger': 'worker-1.test.log',
+            'message': 'a',
+            'level': logging.INFO
+        })
+
+    def test_log_debug(self):
+        '''Logging a debug level log should not generate a log, since it is
+        below the minimum log level.'''
+        self.obs({'message': ["a"], 'system': 'foo',
+                  'logLevel': logging.DEBUG})
+        self.assertEqual(len(self.logfile.logs), 0)
+
+    def test_log_with_context_sentinel(self):
+        '''If the context sentinel has been set for a log, it should not be
+        logged again.'''
+        event = {'message': ["a"], 'system': 'foo'}
+        event.update(self.obs.log_context)
+        self.obs(event)
+        self.assertEqual(len(self.logfile.logs), 0)
+
+
+class TestJunebugLoggerSerivce(VumiTestCase):
+
+    def setUp(self):
+        self.patch(junebug.logging_service, 'LogFile', DummyLogFile)
+        self.logger = LogPublisher()
+        self.service = JunebugLoggerService(
+            'worker-id', '/testpath/', 1000000, 7, logger=self.logger)
+
+    @inlineCallbacks
+    def test_logfile_parameters(self):
+        '''When the logfile is created, it should be created with the correct
+        parameters.'''
+        yield self.service.startService()
+        logfile = self.service.logfile
+        self.assertEqual(logfile.worker_id, 'worker-id')
+        self.assertEqual(logfile.path, '/testpath/')
+        self.assertEqual(logfile.rotateLength, 1000000)
+        self.assertEqual(logfile.maxRotatedFiles, 7)
+
+    @inlineCallbacks
+    def test_logging(self):
+        '''The logging service should write logs to the logfile when the
+        service is running.'''
+        yield self.service.startService()
+        logfile = self.service.logfile
+        self.logger.msg("Hello", logLevel=logging.WARN)
+        [log] = logfile.logs
+        log = json.loads(log)
+        log.pop('timestamp')
+
+        self.assertEqual(log, {
+            'level': logging.WARN,
+            'logger': 'worker-id',
+            'message': 'Hello',
+        })
```

Comment:
Maybe we could factor these extract log, pop timestamp, assert lines out into a helper function?